intellij/src/saros/intellij/editor/EditorAPI.java

```diff
@@ -104,6 +106,43 @@ static TextSelection getSelection(@NotNull Editor editor) {
     return calculateSelectionPosition(editor, selectionStartOffset, selectionEndOffset);
   }
 
+  /**
+   * Returns whether the editor currently has a backwards selection.
+   *
+   * <p>A backwards selection is defined by the start of the selection being located before the end
+   * of the selection. This is checked by using the position of the caret position in the editor.
+   *
+   * @param editor the editor to check
+   * @return whether the editor currently has a backwards selection
+   * @throws IllegalStateException if the caret position matches neither the start nor the end of
+   *     the selection
+   */
+  public static boolean hasBackwardsSelection(@NotNull Editor editor) {
+    SelectionModel selectionModel = editor.getSelectionModel();
+
+    int selectionStartOffset = selectionModel.getSelectionStart();
+    int selectionEndOffset = selectionModel.getSelectionEnd();
+
+    if (selectionStartOffset == selectionEndOffset) {
+      return false;
+    }
+
+    CaretModel caretModel = editor.getCaretModel();
+
+    Caret caret = caretModel.getPrimaryCaret();
+
+    int caretOffset = caret.getOffset();
+
+    if (caretOffset == selectionEndOffset) {
+      return false;
+    } else if (caretOffset == selectionStartOffset) {
+      return true;
+    } else {
+      throw new IllegalStateException(
+          "Encountered caret which is neither at the end nor at the front of the selection");
```

Comment:
Apparently this can actually happen in normal usage (I am not quite sure when/how), so throwing an exception is not the best handling.

I will have to decide on a default handling for such cases.

Reply:
Logged warning instead and returned `false` in such cases.